src/cpp/include/openvino/genai/generation_handle.hpp

```diff
@@ -74,12 +74,14 @@ class OPENVINO_GENAI_EXPORTS GenerationHandleImpl {
 
     bool can_read();
 
+    void drop();
+
     GenerationOutputs back();
     // Reads result of a generation for single iteration
     GenerationOutputs read();
     // Reads all generated tokens for all sequences
     std::vector<GenerationOutput> read_all();
 };
 
-using GenerationHandle = std::unique_ptr<GenerationHandleImpl>;
+using GenerationHandle = std::shared_ptr<GenerationHandleImpl>;
```

Comment:
With that change we can have multiple handles pointing to a single stream and we give an explicit option to drop generation via method call. This means that one handle can call `drop` and invalidate handle for not only itself but potentially other handles.

I think that if we go this way we should block any calls on handle that has been dropped (throw errors for example).

Reply:
I think that it is safe to say that current approach with unique_ptr did not restrict GenAI API users from handle misusage, one could simply take the reference of the handle and do whatever.

Changing to shared_ptr and exposing explicit `drop()` method gives more flexibilty which OVMS needed - dropping the request in HTTP client disconnection callback (look up OVMS pull request). I think this will be clear multiple threads use the handle (http thread and the mediapipe thread) and the generation is dropped once all handle shared references are gone.